io/include/pcl/compression/organized_pointcloud_conversion.h

```diff
@@ -324,7 +324,8 @@ namespace pcl
             if (convertToMono)
             {
               // Encode point color
-              const uint32_t rgb = *reinterpret_cast<const int*> (&point.rgb);
+              uint32_t rgb;
+              std::memcpy (&rgb, &point.rgb, sizeof (point.rgb));
               uint8_t grayvalue = static_cast<uint8_t>(0.2989 * static_cast<float>((rgb >> 16) & 0x0000ff) +
                                                        0.5870 * static_cast<float>((rgb >> 8)  & 0x0000ff) +
                                                        0.1140 * static_cast<float>((rgb >> 0)  & 0x0000ff));
```

Comment:
This also is totally outdated. A point that has color information will always have `rgb`, `rgba`, `r`, `g`, `b`, and `a` fields, all of them. So we can directly write `0.2989 * point.r + ....`